demoScripts/createVM.md

```diff
@@ -0,0 +1,137 @@
+
+The following example uses hard coded values for each of the parameters as one of the acceptable (but not recommended) patterns for an Executable document.
+
+# Quickstart: Create a Linux virtual machine with the Azure CLI
+
+**Applies to:** :heavy_check_mark: Linux VMs
+
+This quickstart shows you how to use the Azure CLI to deploy a Linux virtual machine (VM) in Azure. The Azure CLI is used to create and manage Azure resources via either the command line or scripts.
+
+In this tutorial, we will be installing the latest Debian image. To show the VM in action, you'll connect to it using SSH and install the NGINX web server.
+
+If you don't have an Azure subscription, create a [free account](https://azure.microsoft.com/free/?WT.mc_id=A261C142F) before you begin.
+
+## Launch Azure Cloud Shell
+
+The Azure Cloud Shell is a free interactive shell that you can use to run the steps in this article. It has common Azure tools preinstalled and configured to use with your account. 
+
+To open the Cloud Shell, just select **Try it** from the upper right corner of a code block. You can also open Cloud Shell in a separate browser tab by going to [https://shell.azure.com/bash](https://shell.azure.com/bash). Select **Copy** to copy the blocks of code, paste it into the Cloud Shell, and select **Enter** to run it.
+
+If you prefer to install and use the CLI locally, this quickstart requires Azure CLI version 2.0.30 or later. Run `az --version` to find the version. If you need to install or upgrade, see [Install Azure CLI]( /cli/azure/install-azure-cli).
+
+## Create a resource group
+
+Create a resource group with the [az group create](/cli/azure/group) command. An Azure resource group is a logical container into which Azure resources are deployed and managed. The following example creates a resource group named *myResourceGroup* in the *eastus* location:
+
+```bash
+az group create --name myResourceGroup --location eastus
+```
+
+## Create virtual machine
+
+Create a VM with the [az vm create](/cli/azure/vm) command.
+
+The following example creates a VM named *myVM* and adds a user account named *azureuser*. The `--generate-ssh-keys` parameter is used to automatically generate an SSH key, and put it in the default key location (*~/.ssh*). To use a specific set of keys instead, use the `--ssh-key-values` option.
+
+```bash
+az vm create \
+  --resource-group myResourceGroup \
```

Comment:
We need to discuss this before I approve merge. Can you explain why you believe it would be a good idea to allow hard coded values for elements that will cause errors if there is duplication, such as resource-group name. My concern is that we are allowing docs authors to be lazy at the expense of customers.

From a development perspective this is problematic as we will almost certainly get naming conflicts in tests.

Reply:
I think those are both very valid points. A further point is a document written that way would also break in automated testing for groups as it would surely end with a name conflict. The current mechanism to get past this requires the use of environment variables. 

The reason for adding it was twofold. 

1. I wanted to show the bare minimum a doc writer would need to do in order to transform their documents into executable docs. The hope here was that this would further encourage people to adopt this practice. The hope was eventually to take advantage of automation and testing they would swap it over to env variables. I believe your concern overrides this and you are right that we should require more work upfront for a better customer experience.
2. There is some notion going around that leadership had encouraged doc writers to NOT use env variables as they would lead to customers doing actions which they didn't fully understand. As such env variables are not the most common pattern at the moment.

I will remove the hard coded values example in favor for the best customer experience and to automation within the project examples. I will also dig in a little more into point 2 with various groups to see what the truth is.

Reply:
We should set the bar at what we believe to be acceptable - no hard coded values that should be unique. You will find that docs writers will take the easy route anyway. They will publish with hard coded values and their tests will break or their customers will be unhappy. While some others will adhere to the guidelines and be surprised when others break and customers complain.

By being able to say "we always told you to do it the right way" there is no risk of our work being seen as the cause of the problem (i.e. people can't say "well automated testing is a good idea but it doesn't work in practice" rather they will say "there's a little work needed to truly make it work for you, but it is worth it.)"

As for the LT guidance to not use env variables you are correct. But they are mistaken. This has always been the way to build Linux documentation, it make copy and paste easier and Linux devs are smart-lazy. I'm willing to fight that fight as often as I need to.